### PR TITLE
Configs and environment rework

### LINKED ISSUE #298: Move non-secret chain data away from env to local configs
Currently, [our config](https://github.com/paritytech/polkadot-testnet-faucet/blob/main/env.faucet.config.json#L10-L54) holds data like `FAUCET_BALANCE_CAP`, `INJECTED_TYPES`, `NETWORK_DECIMALS`, `RPC_ENDPOINT`, `DRIP_AMOUNT`.  
Updating such network-specific data is tedious and in many cases requires changing both code and deployment configuration.  

As all this data isn't secret and basically provides configuration for specific network for the faucet, I suggest we do something like [networkData.ts](https://github.com/paritytech/polkadot-testnet-faucet/blob/main/client/src/lib/utils/networkData.ts) (or even reuse that one), so the deployment configuration would only have network name, and everything else would be picked from code.  

Also, `SMF_BOT` and `SMF_BACKEND` prefixes are kinda outdated as they are integrated now. Would be great to discontinue them at some point.

### PR DESCRIPTION
**Pull request by mutantcornholio**
1. Merging configs for BOT and SERVER together. As the faucet is now only one application, the division brings more trouble than it solves
2. Network configuration data now lives in src/networkData.ts. This allows us to add new configuration options without updating whole deployment pipeline in three places.
3. Dropped explicit `validateConfig` invocation. Now config is validated when module is imported during startup.
4. As I traced "dripAmount" configuration down to its usages, I've discovered double conversions like "bigint->float->bigint", which cancels whole idea of using bigint in order to escape rounding errors.  
Reworked it to operate only on bigint, and use conversion only for output.  
5. For some reason, e2e tests didn't want to work on HTTP endpoints anymore (probably, local environment issues). Changed it to WS as it should've been done a long time ago.

IMPORTANT: this PR will have its counterparts in helm-charts and cloud-infra. Do not merge unless following changes are merged first:
https://github.com/paritytech/helm-charts/pull/257
https://gitlab.parity-mgmt.parity.io/infrastructure/cloud-infra/-/merge_requests/1481

Fixes #298

### CODE REVIEW

**Review by rzadp — APPROVED** [review]
Great cleanup, I'm loving it! Few nitpicks for consideration.

File: src/networkData.ts [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+  id: number;
+}
+
+export interface NetworkData {
+  networkName: string;
+  currency: string;
+  chains: ChainData[];
+  explorer: string | null;
+  rpcEndpoint: string;
+  injectedTypes: Record<string, string>;
```
[the remark this answers is not in the document]
Comment (mutantcornholio) [reply src/networkData.ts:14]: I was also thinking about it. Let's do it.

File: env.faucet.config.json [hunk 2]
```diff
@@ -42,34 +22,27 @@
...
         "type": "string"
       },
-      "DRIP_AMOUNT": {
-        "description": "default amount of token to send",
-        "default": 0.5,
-        "type": "number"
-      },
       "MATRIX_ACCESS_TOKEN": {
         "description": "your bot access token here is how to find it https://t2bot.io/docs/access_tokens/.",
         "required": true,
```
[the remark this answers is not in the document]
Comment (mutantcornholio) [reply env.faucet.config.json:48]: Is it though? Here how the startup looks with `SMF_CONFIG_NETWORK` not being set first with `mandatory: true`, and then without this property:
Screen Shot 2023-06-19 at 14 01 30
Screen Shot 2023-06-19 at 14 01 33

[the remark this answers is not in the document]
Comment (rzadp) [reply env.faucet.config.json:48]: @mutantcornholio I'm talking about `"required": true`, not `"mandatory": true`

[the remark this answers is not in the document]
Comment (mutantcornholio) [reply env.faucet.config.json:48]: Ahh...
At first I thought that we're just using an incorrect option, not mixing two of them. Yeah, totally makes sense to remove it, as adding "mandatory" there would actually break deployment for rococo web faucet (which is not yet merged with rococo matrix faucet)

File: env.faucet.config.json [hunk 3]
```diff
@@ -42,43 +22,34 @@
         "default": "unset",
         "type": "string"
       },
+      "PORT": {
+        "description": "the port you want the server to listen on (should be part of the bot's `SMF_CONFIG_BACKEND_URL`.)",
+        "default": 5555,
+        "type": "number"
+      },
       "EXTERNAL_ACCESS": {
```
Comment (mordamax) [inline env.faucet.config.json:30]: @mutantcornholio i don't see this config is somewhere used in https://gitlab.parity-mgmt.parity.io/infrastructure/cloud-infra/-/merge_requests/1481/diffs

do we still need this?

Comment (mutantcornholio) [reply env.faucet.config.json:30]: `EXTERNAL_ACCESS`? Yeah, I'm eager to kill it, but not right now

**Pull request merged**